bundles/org.openhab.core.audio/src/main/java/org/openhab/core/audio/internal/javasound/JavaSoundAudioSource.java

```diff
@@ -88,28 +108,107 @@ public synchronized AudioStream getInputStream(AudioFormat expectedFormat) throw
+    }
+
+    private void startPipeWrite() {
+        if (this.pipeWriteTask == null) {
+            this.pipeWriteTask = executor.submit(() -> {
+                int lengthRead;
+                byte[] buffer = new byte[1024];
+                int readRetries = 3;
+                while (!openStreamRefs.isEmpty()) {
+                    var stream = this.microphone;
+                    if (stream != null) {
+                        try {
+                            lengthRead = stream.read(buffer, 0, buffer.length);
+                            for (var output : openStreamRefs) {
+                                try {
+                                    output.write(buffer, 0, lengthRead);
+                                    if (openStreamRefs.contains(output)) {
+                                        output.flush();
+                                    }
+                                } catch (IOException e) {
+                                    if (e instanceof InterruptedIOException && openStreamRefs.isEmpty()) {
+                                        // task has been ended while writing
+                                        return;
+                                    }
+                                    LOGGER.warn("IOException while writing to source pipe: {}", e.getMessage());
+                                } catch (RuntimeException e) {
+                                    LOGGER.warn("RuntimeException while writing to source pipe: {}", e.getMessage());
+                                }
+                            }
+                        } catch (RuntimeException e) {
+                            LOGGER.warn("RuntimeException while reading from pulse source: {}", e.getMessage());
+                        }
+                    } else {
+                        LOGGER.warn("Unable access to microphone stream");
+                    }
+                }
+                this.pipeWriteTask = null;
+            });
+        }
+    }
+
+    private void unregisterPipe(PipedOutputStream pipeOutput) {
+        synchronized (openStreamRefs) {
+            openStreamRefs.remove(pipeOutput);
+            try {
+                Thread.sleep(0);
+            } catch (InterruptedException ignored) {
+            }
+            stopPipeWriteTask();
+            try {
+                pipeOutput.close();
+            } catch (IOException ignored) {
+            }
+        }
+    }
+
+    private void stopPipeWriteTask() {
+        synchronized (openStreamRefs) {
+            var microphone = this.microphone;
+            if (openStreamRefs.isEmpty()) {
+                if (pipeWriteTask != null) {
+                    pipeWriteTask.cancel(true);
+                    this.pipeWriteTask = null;
+                }
+                if (microphone != null) {
                     microphone.close();
                     this.microphone = null;
                 }
             }
-        });
+        }
```

Comment:
Since this is only called from l. 190, wouldn't it be better to integrate it there? If you want to keep it as a separate method, please chose a different name. The method does not "stop" the task at any time, it includes a check if it is necessary.